addon-chrome/db.js

```diff
@@ -0,0 +1,118 @@
+const DB_NAME = "activitystream";
+const DB_MODE = "readwrite";
+let _db = null;
+
+module.exports = class Db {
+	static init(keyStores) {
+		const initPromise = new Promise((resolve, reject) => {
+			const request = indexedDB.open(DB_NAME, 1);
+
+			request.onerror = function(event) {
+				throw new Error("IndexedDB not available?!");
+			};
+
+			request.onsuccess = function(event) {
+				if (event.target.result.objectStoreNames.length > 0) {
+					_db = event.target.result;
+					resolve();
+				}
+			};
+
+			request.onupgradeneeded = function(event) {
+				_db = event.target.result;
+				const stores = [];
+				for(const store in keyStores) {
+					const keyPath = keyStores[store].keyPath;
+					stores.push(store);
+					const objectStore = _db.createObjectStore(store, {keyPath});
+					const index = keyStores[store].index;
+					if (index) {
```

Comment:
nit: i'd be more explicit with checks such as these. e.g. do index === undefined instead. index="false" or "0" will screw this up